sdk/loadtestservice/load-testing-rest/review/load-testing.api.md

```diff
@@ -9,433 +9,309 @@
 import { HttpResponse } from '@azure-rest/core-client';
+import { OperationState } from '@azure/core-lro';
 import { PagedAsyncIterableIterator } from '@azure/core-paging';
 import { PathUncheckedResponse } from '@azure-rest/core-client';
 import { RawHttpHeaders } from '@azure/core-rest-pipeline';
 import { RequestParameters } from '@azure-rest/core-client';
+import { SimplePollerLike } from '@azure/core-lro';
 import { StreamableMethod } from '@azure-rest/core-client';
 import { TokenCredential } from '@azure/core-auth';
 
-// @public (undocumented)
+// @public
 export interface AppComponent {
     displayName?: string;
     kind?: string;
     resourceGroup?: string;
-    resourceId: string;
-    resourceName: string;
-    resourceType: string;
+    resourceId?: string;
+    resourceName?: string;
+    resourceType?: string;
     subscriptionId?: string;
 }
 
-// @public (undocumented)
-export interface AppComponentCreateOrUpdateAppComponents {
-    delete(options?: AppComponentDeleteAppComponentsParameters): StreamableMethod<AppComponentDeleteAppComponents204Response | AppComponentDeleteAppComponentsdefaultResponse>;
-    get(options?: AppComponentGetAppComponentByNameParameters): StreamableMethod<AppComponentGetAppComponentByName200Response | AppComponentGetAppComponentByNamedefaultResponse>;
-    patch(options: AppComponentCreateOrUpdateAppComponentsParameters): StreamableMethod<AppComponentCreateOrUpdateAppComponents200Response | AppComponentCreateOrUpdateAppComponents201Response | AppComponentCreateOrUpdateAppComponentsdefaultResponse>;
-}
-
-// @public
-export interface AppComponentCreateOrUpdateAppComponents200Response extends HttpResponse {
-    // (undocumented)
-    body: AppComponentsMapOutput;
-    // (undocumented)
-    status: "200";
-}
-
-// @public
-export interface AppComponentCreateOrUpdateAppComponents201Response extends HttpResponse {
-    // (undocumented)
-    body: AppComponentsMapOutput;
-    // (undocumented)
-    status: "201";
-}
-
-// @public (undocumented)
-export interface AppComponentCreateOrUpdateAppComponentsBodyParam {
-    body: AppComponentsMap;
-}
-
-// @public (undocumented)
-export interface AppComponentCreateOrUpdateAppComponentsdefaultHeaders {
-    "x-ms-error-code"?: string;
-}
-
 // @public
-export interface AppComponentCreateOrUpdateAppComponentsdefaultResponse extends HttpResponse {
-    // (undocumented)
-    body: ErrorResponseBodyOutput;
-    // (undocumented)
-    headers: RawHttpHeaders & AppComponentCreateOrUpdateAppComponentsdefaultHeaders;
-    // (undocumented)
-    status: string;
-}
-
-// @public (undocumented)
-export interface AppComponentCreateOrUpdateAppComponentsMediaTypesParam {
-    contentType?: "application/merge-patch+json";
+export interface AppComponentOutput {
+    displayName?: string;
+    kind?: string;
+    resourceGroup?: string;
+    resourceId?: string;
+    resourceName?: string;
+    resourceType?: string;
+    subscriptionId?: string;
```

Comment:
same here, are all of these really optional?

Reply:
Generated by auto-rest.